### PR TITLE
Issue 912: il_arts_council

### PR DESCRIPTION
**Pull request by channiemills**
PR for https://github.com/City-Bureau/city-scrapers/issues/912. 

I did have a question, how should I handle `cancelled` meetings? The [website](http://www.arts.illinois.gov/about-iac/governance/council-meetings) uses the location field to indicate whether a meeting is canceled or not but I wasn't sure if there was a different field I should use on the meeting object to indicate that.

### CODE REVIEW

**Comment by pjsier** [comment]
@channiemills thanks for the PR! You can pass the string of the location (or wherever might mention a cancellation) to the `_get_status` function in the `text` kwarg. More info here https://cityscrapers.org/docs/development/#status

It looks like the check for `isort` is failing for now though, but running the linting commands locally should fix that

File: city_scrapers/spiders/il_arts_council.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                meeting["status"] = self._get_status(meeting)
+                meeting["id"] = self._get_id(meeting)
+
+                yield meeting
+
+    def _parse_title(self, item):
+        """Parse or generate meeting title."""
+        return "Board Meeting"
+
+    def _parse_description(self, item):
```
Comment (pjsier) [inline city_scrapers/spiders/il_arts_council.py]: We can leave this blank if they aren't providing any info in the description

File: city_scrapers/spiders/il_arts_council.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                )
+
+                meeting["status"] = self._get_status(meeting)
+                meeting["id"] = self._get_id(meeting)
+
+                yield meeting
+
+    def _parse_title(self, item):
+        """Parse or generate meeting title."""
+        return "Board Meeting"
```
Comment (pjsier) [inline city_scrapers/spiders/il_arts_council.py]: "Agency Board" might be the best default here. We try to leave "Meeting" out if we can

File: city_scrapers/spiders/il_arts_council.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    def _parse_location(self, item):
+        """Parse or generate location."""
+
+        location = item.xpath('td[2]/text()').get()
+        if not location:
+            location = item.xpath('td[2]/a/text()').get()
+
+        jrtc = {
+            "address":
+                "James R. Thompson Center, 100 West Randolph, Suite 10-500, Chicago, IL 60601",  # noqa
```
Comment (pjsier) [inline city_scrapers/spiders/il_arts_council.py]: We can take out the "James R. Thompson Center" part here and just start with "100 W"

File: city_scrapers/spiders/il_arts_council.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                "name": "",
+            }
+
+    def _parse_links(self, item):
+        """Parse or generate links."""
+        agenda_link = item.xpath('td/a/@href').get()
+        if agenda_link:
+            title = agenda_link.split('/')[-1].replace('%20', ' ')
+            return [{"href": agenda_link, "title": title}]
+        return [{"href": "", "title": ""}]
```
Comment (pjsier) [inline city_scrapers/spiders/il_arts_council.py]: If there aren't any links available this should just return an empty list

**Comment by channiemills** [comment]
Thanks for the feedback, back to you @pjsier

File: city_scrapers/spiders/il_arts_council.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+        else:
+            return {
+                "address": "",
+                "name": "",
+            }
+
+    def _parse_links(self, item):
+        """Parse or generate links."""
+        agenda_link = item.xpath('td/a/@href').get()
```
Comment (pjsier) [inline city_scrapers/spiders/il_arts_council.py]: I missed this on the first pass, but it looks like some of the URLs being returned are relative (seeing it on the August 2018 meeting). To be safe, we should add `response` as an argument to this function and call `response.urljoin` on the parsed link to make sure that it includes the domain

File: tests/test_il_arts_council.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    assert parsed_items[0]["location"] == {'address': 'TBA', 'name': 'TBA'}
+
+
+def test_source():
+    assert parsed_items[0][
+        "source"] == "http://www.arts.illinois.gov/about-iac/governance/council-meetings"  # noqa
+
+
+def test_links():
+    print(parsed_items[8]["links"])
```
Comment (pjsier) [inline tests/test_il_arts_council.py]: This can be removed

Comment (channiemills) [reply tests/test_il_arts_council.py]: 🙈

File: city_scrapers/spiders/il_arts_council.py (hunk 5 above)
Comment (channiemills) [reply city_scrapers/spiders/il_arts_council.py]: Good catch, thanks.

**Comment by pjsier** [comment]
@channiemills looks great, thanks! Merging now

**Pull request merged**